### PR TITLE
Refactoring the StepWalker to support 0.0 and float(inf)

### PR DESCRIPTION
**Pull request by th3w4y**
Fixes:
- support for 0.0 and infinite values as speed of StepWalker
- follow_path forced to use StepWalker and to walk... 
- changed default polylinewalker speed to a random uniform value between self.bot.config.walk_min and self.bot.config.walk_max

### CODE REVIEW

**Comment by th3w4y** [comment]
@mjmadsen Can you please run some tests  in case you  are merging this to your branch

I did not manage to do that...

I have to go to sleep...

**Comment by mjmadsen** [comment]
Haha, yeah I can when I get home from work. Have a good rest man.

**Comment by th3w4y** [comment]
@mjmadsen since is a complete refactor using a different method to calculate the locations. all the tests in step_walker_test.py are failing... 

maybe you can checkout my branch and see the behaviour..

**Comment by mjmadsen** [comment]
I'll be home in about 4 hours then I can test. You're rocking the changes!

File: pokemongo_bot/walkers/step_walker.py [hunk 1]
```diff
@@ -1,106 +1,85 @@
...
-
-        self.destLat = dest_lat
-        self.destLng = dest_lng
-        self.totalDist = max(1, self.dist)
-
-        if self.speed == 0:
-            raise Exception("Walking speed cannot be 0, change your walking speed higher than 1!")
-        else:
-            self.steps = (self.dist + 0.0) / (self.speed + 0.0)
+            # Use this to pass custom values from 0.0 to stay put to float("inf") to teleport
```
Comment (mjmadsen) [inline pokemongo_bot/walkers/step_walker.py]: Do know there was a lot of rub about very low speeds when we did some walker work before as the step size calculations we did before resulted in the step() being called forever.

**Pull request closed without merging**